sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -654,7 +660,313 @@ void setAndGetSessionState(MessagingEntityType entityType) {
     }
 
     /**
-     * Sets the sender and receiver. If session is enabled, then a single-named session receiver is created.
+     * Verifies that we can not receive from a session enabled receiver which try to connect to non session entity.
+     */
+    @MethodSource("messagingEntityProvider")
+    @ParameterizedTest
+    void createSessionReceiverForNonSessionEntity(MessagingEntityType entityType) {
+        // Arrange
+        String sessionId = "test-session-id";
+        ServiceBusClientBuilder.ServiceBusSessionReceiverClientBuilder builder = getBuilder(false)
+            .sessionReceiver();
+
+        if (entityType == MessagingEntityType.QUEUE) {
+            builder.queueName(getQueueName());
+        } else if (entityType == MessagingEntityType.SUBSCRIPTION) {
+            builder.topicName(getTopicName()).subscriptionName(getSubscriptionName());
+        }
+
+        ServiceBusReceiverAsyncClient invalidReceiver = builder.sessionId(sessionId).buildAsyncClient();
+
+        // Assert & Act
+        StepVerifier.create(invalidReceiver.receive())
+            .verifyErrorMatches(error -> error instanceof UnsupportedOperationException);
+    }
+
+
+    /**
+     * Send multiple message and receive from multiple receivers and autocomplete is off.
+     */
+    @Disabled("Last message does not arrive : https://github.com/Azure/azure-sdk-for-java/issues/10840")
+    @MethodSource("messagingEntityWithSessions")
+    @ParameterizedTest
+    void multipleReceiverAndClientComplete(MessagingEntityType entityType, boolean isSessionEnabled) {
+        // Arrange
+        setSenderAndReceiver(entityType, isSessionEnabled);
+
+        int firstBatch = 2;
+        int secondBatch = 2;
+        final String messageId = UUID.randomUUID().toString();
+        final List<ServiceBusMessage> messages;
+        if (isSessionEnabled) {
+            messages = TestUtils.getServiceBusMessages(firstBatch, CONTENTS_BYTES, messageId, sessionId);
+            messages.addAll(TestUtils.getServiceBusMessages(secondBatch, CONTENTS_BYTES, messageId, sessionId2));
+        } else {
+            messages = TestUtils.getServiceBusMessages(firstBatch + secondBatch, messageId,
+                CONTENTS_BYTES);
+        }
+
+        final Duration smallDuration = Duration.ofSeconds(5);
+        final ReceiveAsyncOptions options = new ReceiveAsyncOptions().setIsAutoCompleteEnabled(false);
+
+        sendMessage(messages).block(TIMEOUT);
+
+        // Assert & Act
+        StepVerifier.create(receiver.receive(options))
+            .assertNext(message -> {
```

Comment:
nit: context instead of message. In subsequent calls you do `message.getMessage()` which is misleading.